### PR TITLE
Add providerID not set issue and fix to troubleshooting guide

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/troubleshooting/troubleshooting.md [hunk 1]
```diff
@@ -439,6 +439,27 @@ Status:
   Phase:                   Running
 ```
 
+### Machine gets stuck at `Provisioned` state without `providerID` set
+
+The VM can be created in the provider infrastructure with proper IP assigned but running `kubectl get machines -n eksa-system` indicates that the machine is in `Provisioned` state and never gets to `Running`.
+
+Check the CAPI controller manager log with `kubectl logs -f capi-controller-manager-7f754cf76b-g92ht -n capi-system`:
```
Comment (jaxesn) [inline docs/content/en/docs/troubleshooting/troubleshooting.md:446]: how do we usually show things like this where the deployment name will be somewhat random? do we just assume they know to find their specific one?

Comment (jiayiwang7) [reply docs/content/en/docs/troubleshooting/troubleshooting.md]: updated to use label filter

**Review by jaxesn — APPROVED** [review]
/approve

**Pull request merged**